core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala

```diff
@@ -17,9 +17,14 @@ import scala.jdk.CollectionConverters._
 trait BillingProfileManagerDAO {
   def createBillingProfile(displayName: String, billingInfo: Either[RawlsBillingAccountName, AzureManagedAppCoordinates], userInfo: UserInfo): Future[ProfileModel]
 
-  def listBillingProfiles(samUserResources: Seq[SamUserResource], userInfo: UserInfo)(implicit ec: ExecutionContext): Future[Seq[RawlsBillingProject]]
+  /***
+    * Uses BPM to retrieve billing profile information for each of the RawlsBillingProjects (via their billingProjectIDs).
+    */
+  def populateBillingProfiles(samUserResources: Seq[SamUserResource], userInfo: UserInfo, rawlsBillingProjects: Seq[RawlsBillingProject])(implicit ec: ExecutionContext): Future[Seq[RawlsBillingProject]]
 
   def listManagedApps(subscriptionId: UUID, userInfo: UserInfo): Future[Seq[AzureManagedAppModel]]
+
+  def getBillingProfile(billingProfileId: UUID, userInfo: UserInfo): ProfileModel
```

Comment:
I noticed that `createBillingProfile` returns `Future[ProfileModel]`, but I don't understand why. I would imagine that we want both methods to return the same thing, but adding in the `Future` seems unnecessarily complex to me.

@aherbst-broad what was your thinking with wrapping the return value in a `Future.successful`?

Reply:
It also doesn't look like `listManagedApp` actually needs to return a Future… I feel like I'm missing something.